samples/logging_s3/pipeline_log_to_s3_by_fluentd_recommend/fluent-config.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: fluent-config
+  namespace: kube-system
+data:
+  fluent.conf: |-
+    <source>
+      @type tail
+      path "/var/log/containers/*.log"
+      pos_file /var/log/fluentd-docker.pos
```

Comment:
@fenglixa Can you explain what this pos_file is going to do? I think we are having some issue that's specific on IBM Cloud where we can see the logs under `/var/log/containers/*.log` and `/var/log/fluentd-docker.pos` in the daemonset, but the source is not returning anything despite I matched them with `**` wildcard. Is there a way to show the debugging logs since fluentd itself doesn't have any error.

On minikube it works perfectly fine. What I can do is merge this PR after you fix the configmap name, then we can share with you an ibm cloud shared vm environment to figure out what changes need to make for the fluentd source on our IBM Cloud env.

Reply:
@Tomcli, this is just an example, user need to config it based on their real situation.  I already leaved the interface for them to config it. The pos_file was put here to avoid logging duplicate after delete/recreate again.  

The best solution is leave a special mountpoint for puting pos_file, but I would like leave it to user to do it.
How do you think?

Reply:
I have a quick test on IBM Cloud (Openshift Cluster). No error found. As below for details:
```
 fengli@fenglis-mbp> oc get po -o wide -n kube-system
NAME                             READY   STATUS    RESTARTS   AGE     IP             NODE                           NOMINATED NODE   READINESS GATES
fluentd-s3-c287g                 1/1     Running   0          6m25s   10.254.8.147   worker0.feng.os.fyre.ibm.com   <none>           <none>
fluentd-s3-rpq55                 1/1     Running   0          6m25s   10.254.4.40    worker1.feng.os.fyre.ibm.com   <none>           <none>
fluentd-s3-tk28f                 1/1     Running   0          6m25s   10.254.12.76   worker2.feng.os.fyre.ibm.com   <none>           <none>
tiller-deploy-589d7d9489-b5v74   1/1     Running   1          3d4h    10.254.8.35    worker0.feng.os.fyre.ibm.com   <none>           <none>

 fengli@fenglis-MacBook-Pro> oc exec -it -n kube-system fluentd-s3-ggl55 -- sh
/ # ls /var/log/fluentd-docker.pos
/var/log/fluentd-docker.pos
/ # cat /var/log/fluentd-docker.pos
/var/log/containers/csi-rbdplugin-dmb82_rook-ceph_driver-registrar-3389f844e1553ca08b96e5b07d47b329156d7af006ebec2a821d35d1c112004e.log	00000000000008e5	0000000004a02d1c
/var/log/containers/ovs-nxlf2_openshift-sdn_openvswitch-63bfb792f508cdda77a562dcec1c67ab52fdaf1e435e2ca52a49b42463ed6e07.log	000000000001950f	00000000056000ed
... 

 fengli@fenglis-MacBook-Pro> oc logs -n kube-system fluentd-s3-v5dx8
 fengli@fenglis-MacBook-Pro>
```

@Tomcli , FYI...

Reply:
It could be problem for the shared vm on ibm public cloud. Fyre usually has dedicated nodes for creating openshift so it might not have this problem.

On my fluentd agent, the pos_file works. However, it looks like fluentd aren't able to parse those files to s3 probably due to similar the issue we had with the minio client. Here is the errors I got after running it for few hours.
```shell
2020-08-17 14:51:53 +0000 [error]: Exception encountered parsing namespace watch event. The connection might have been closed. Retried 10 times yet still failing. Restarting.error reading from socket: Could not parse data
#<Thread:0x000055917f00cba0@/usr/lib/ruby/gems/2.5.0/gems/fluent-plugin-kubernetes_metadata_filter-2.5.2/lib/fluent/plugin/filter_kubernetes_metadata.rb:279 run> terminated with exception (report_on_exception is true):
/usr/lib/ruby/gems/2.5.0/gems/fluent-plugin-kubernetes_metadata_filter-2.5.2/lib/fluent/plugin/kubernetes_metadata_watch_namespaces.rb:70:in `rescue in set_up_namespace_thread': Exception encountered parsing namespace watch event. The connection might have been closed. Retried 10 times yet still failing. Restarting. (Fluent::UnrecoverableError)
	from /usr/lib/ruby/gems/2.5.0/gems/fluent-plugin-kubernetes_metadata_filter-2.5.2/lib/fluent/plugin/kubernetes_metadata_watch_namespaces.rb:39:in `set_up_namespace_thread'
	from /usr/lib/ruby/gems/2.5.0/gems/fluent-plugin-kubernetes_metadata_filter-2.5.2/lib/fluent/plugin/filter_kubernetes_metadata.rb:279:in `block in configure'
Unexpected error Exception encountered parsing namespace watch event. The connection might have been closed. Retried 10 times yet still failing. Restarting.
  /usr/lib/ruby/gems/2.5.0/gems/fluent-plugin-kubernetes_metadata_filter-2.5.2/lib/fluent/plugin/kubernetes_metadata_watch_namespaces.rb:70:in `rescue in set_up_namespace_thread'
  /usr/lib/ruby/gems/2.5.0/gems/fluent-plugin-kubernetes_metadata_filter-2.5.2/lib/fluent/plugin/kubernetes_metadata_watch_namespaces.rb:39:in `set_up_namespace_thread'
  /usr/lib/ruby/gems/2.5.0/gems/fluent-plugin-kubernetes_metadata_filter-2.5.2/lib/fluent/plugin/filter_kubernetes_metadata.rb:279:in `block in configure'
```

And in the pod I'm able to see the pos_file working fine:
```
base) tommys-mbp:pipeline_log_to_s3_by_fluentd_recommend tommyli$ kubectl exec -ti fluentd-s3-4pjq5 -n kube-system /bin/sh
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl kubectl exec [POD] -- [COMMAND] instead.
/ # cat /var/log/fluentd-docker.pos
/var/log/containers/catalog-operator-67cbc5d959-wbhk2_ibm-system_catalog-operator-6baf9c00de74532ad5431c7e3f84e3bdebc8e44a0bd8e1018ece55c053c88707.log	0000000000002b0a	0000000000540027
/var/log/containers/conditional-execution-pipeline-d4adf-print-4-sl9lq-super--2prft_tommy-chaoping-li_place-scripts-c5846845e5b69b09ce992c701dc09dc623a167b0794bf62895712f96eb0f0b6b.log	0000000000000000	00000000005809d5
/var/log/containers/conditional-execution-pipeline-bd2d9-print-4-9b96g-pod-6mpsq_tommy-chaoping-li_step-main-fa63062bdff439694de5f13942c133ef78fcd66fdaf63263870e4aaed90675d0.log	0000000000000065	00000000005809ce
/var/log/containers/istio-nodeagent-t8zl5_istio-system_nodeagent-2e87d9d8ffe69627223159df19283cfb3a498b0f1210a2de6127d26ee9b6de8a.log	00000000000042e6	0000000000580607
/var/log/containers/conditional-execution-pipeline-d4adf-print-3-nwf56-super--zvmlj_tommy-chaoping-li_place-scripts-7c1e22643c90f014e86b550d31f36526088c4c0c78e344750cd136a37ba3547a.log	0000000000000000	00000000005809d8
/var/log/containers/networking-istio-6dcd5c7f7f-q8md5_knative-serving_networking-istio-9cda417d131df6b2f42cc756b09a8c050d70efe427b83104eb14d4aa8c649fd2.log	000000000002669c	0000000000580855
/var/log/containers/calico-node-ndbs2_kube-system_calico-node-97247f335eab25570d045b67409e67114627832b30885708a8393af413076b46.log	0000000002739843	000000000054001b
/var/log/containers/metadata-db-595fcfb899-7bw9f_kubeflow_db-container-974c298cc54f9311b5041f6d5bb739a265424aa8f3239d188f66273eb5eaf60c.log	0000000000006008	000000000058085c
/var/log/containers/conditional-execution-pipeline-bd2d9-generate-random-numb-wm9gh_tommy-chaoping-li_step-condition-check-super-condition-582c102a1ebdadc5a004374ed70aabdbac1a23be7f7d356019d1476cd3eba738.log	0000000000000271	00000000005809b2
```

Reply:
Thanks @Tomcli, let's make this PR go first, I logged bug #276 to trace the issue you found on ibm cloud, if possible, could you share your env to me?

Reply:
@fenglixa I sent you the cluster information in slack. Let me know if you couldn't access the cluster.